HLM-311 | Adding a consumer for Registry Async Behaviour

ajay-beehyv opened the pull request:
@prasanna-egov @saiprakash-v @egov-joy

prasanna-egov commented on backend/registry-async-handler/main.js:
```diff
@@ -0,0 +1,36 @@
+
+async function init_handler(conf){
+    const kafka = new Kafka({
+        clientId: 'async-handler',
```
Registry Async Handler please

prasanna-egov commented on backend/registry-async-handler/main.js:
```diff
@@ -0,0 +1,36 @@
+    init_postCreateEntity_consumer(conf,kafka);
+}
+
+async function init_postCreateEntity_consumer(conf,kafka){
```
be consistent either use camelCase or _, don't mix

prasanna-egov commented on backend/registry-async-handler/main.js:
```diff
@@ -0,0 +1,36 @@
+    eachMessage: async ({ message}) => {
+      const certificateOsidMessage = JSON.parse(message.value.toString());
+      console.log( {entitytype: certificateOsidMessage.entityType});
+      if(certificateOsidMessage.entityType == conf.CERTIFICATE_ENTITY_TYPE){
```
Where is this conf from

prasanna-egov commented on backend/registry-async-handler/main.js:
```diff
@@ -0,0 +1,36 @@
+      console.log( {entitytype: certificateOsidMessage.entityType});
+      if(certificateOsidMessage.entityType == conf.CERTIFICATE_ENTITY_TYPE){
+        try{
+          const certificateOsidMapResponse = await  axios.post(`${conf.SUNBIRD_CERTIFICATE_URL}${entityType}`, JSON.parse(message.value.toString()),
```
move to services folder. Follow similar forlder structure as other microservices.

prasanna-egov commented on backend/registry-async-handler/package.json:
```diff
@@ -0,0 +1,23 @@
+  "author": "divoc",
+  "license": "GPL-3.0-or-later",
+  "dependencies": {
+    "axios": "^0.21.1",
```
use latest version of axios

prasanna-egov reviewed (changes requested):
Docker compose entry is missing. Dockerfile and Makefile are missing. Add those as well.

egov-joy commented:
@ajay-beehyv please mention the story number in the commit message

prasanna-egov commented:
There are a lot of changes that are needed. Env variables need to move to config file. Main file need not have a module exports. I am sure this entire thing is not working. Could you do dev testing before raising the PR.

ajay-beehyv commented:
@prasanna-egov @egov-joy @saiprakash-v 
 This PR containes registry-async-handler microservice, schema for transactionOsidMap, APi to get osid from tranactionId in vc-management-service.
Removed divoc listener from this PR to reduce complexity and added new issue for it (HLM-477)

saiprakash-v commented on backend/vc-management-service/src/configs/config.js:
```diff
@@ -6,6 +6,7 @@ const JWT_PUBLIC_KEY = process.env.AUTH_PUBLIC_KEY;
 const BASE_URL = process.env.ISSUER_SERVICE_BASE_URL || '/vc-management/';
 const REQUEST_TIMEOUT = process.env.REQUEST_TIMEOUT || 10000;
 const ISSUER_NAME = process.env.ISSUER_NAME || 'Issuer';
+const STORED_ENTITY_TYPE = process.env.STORED_ENTITY_TYPE ;
```
This is just a table name. Can't we have it as a constant and not config?

prasanna-egov commented on backend/vc-management-service/src/services/sunbird.service.js:
```diff
@@ -86,10 +86,31 @@ const getIssuerId = async(token) => {
+    console.log({transctionUrl: constants.SUNBIRD_GET_TRANSACTION_URL, 
+                transactionId: transactionId});
+    const transactionRequest = {
+        id:  "sunbird-rc.registry.search",
```
Move to constants

prasanna-egov commented on backend/vc-management-service/src/services/sunbird.service.js:
```diff
@@ -86,10 +86,31 @@ const getIssuerId = async(token) => {
+    const transactionRequest = {
+        id:  "sunbird-rc.registry.search",
+        ver: "1.0",
+        ets: "",
```
What is this ets.

ajay-beehyv replied:
used that approach fron divoc/certificate-api, I'll test if all those variable are necessary

ajay-beehyv replied:
Removed unnecessary field from Transaction Request

ajay-beehyv commented:
> Docker compose entry is missing. Dockerfile and Makefile are missing. Add those as well.

resolved
